src/main/java/org/cactoos/list/FirstOf.java

```diff
@@ -40,23 +42,57 @@
     /**
      * Source iterable.
      */
-    private final Iterable<T> source;
+    private final Iterable<T> src;
+
+    /**
+     * Fallback value.
+     */
+    private final Scalar<T> fbk;
+
+    /**
+     * Ctor.
+     *
+     * @param src Iterable
+     */
+    public FirstOf(final Iterable<T> src) {
+        this(
+            src,
+            () -> {
+                throw new IOException("Iterable is empty");
+            }
+        );
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param src Iterable
+     * @param fbk Fallback value
+     */
+    public FirstOf(final Iterable<T> src, final T fbk) {
+        this(src, () -> fbk);
+    }
 
     /**
      * Ctor.
      *
-     * @param source Iterable
+     * @param src Iterable
+     * @param fbk Fallback value
      */
-    public FirstOf(final Iterable<T> source) {
-        this.source = source;
+    public FirstOf(final Iterable<T> src, final Scalar<T> fbk) {
```

Comment:
@g4s8 looks good, but I feel that we're not using `Scalar` the right way here. We're losing the context. This scalar won't have any idea of why it's called and what is the situation there. I would rather use `Func<Iterable<T>, T>`. This is how that `Func` will be able to throw a more meaningful exception.